typing/typemod.ml

```diff
@@ -844,11 +944,25 @@ and transl_signature env sg =
             mksig (Tsig_modtype mtd) env loc :: trem,
             sg :: rem,
             final_env
-        | Psig_open sod ->
-            let (_path, newenv, od) = type_open env sod in
+        | Psig_open sod -> begin
+            let (newenv, od) = type_open env sod in
             let (trem, rem, final_env) = transl_sig newenv srem in
+            let remr = ref rem in
+            (match extract_open od with
```

Comment:
I find this (extract_open and the "stack") a bit ugly.  Couldn't type_open return directly the required information?

Reply:
removed stack now.